decidim-core/app/helpers/decidim/layout_helper.rb

```diff
@@ -25,10 +25,11 @@ def apple_favicon
     end
 
     def legacy_favicon
-      icon_image = current_organization.attached_uploader(:favicon).variant_url(:small, host: current_organization.host)
+      variant = :favicon if current_organization.favicon.content_type != "image/vnd.microsoft.icon"
+      icon_image = current_organization.attached_uploader(:favicon).variant_url(variant, host: current_organization.host)
       return unless icon_image
 
-      favicon_link_tag(icon_image.gsub(".png", ".ico"), rel: "icon", sizes: "any", type: nil)
+      favicon_link_tag(icon_image, rel: "icon", sizes: "any", type: nil)
```

Comment:
maybe here, we could use a hard coded `/favicon.ico` ?

Reply:
Few lines above there is the favicon method, which may be changed to set the path to `/favicon.ico` ?

Reply:
No, I prefer to render the Rails ActiveStorage redirect URL here. It performs better.

The `/favicon.ico` path should be only considered as a fallback option that will be used when the browser tries to find the favicon by itself (e.g. when a PDF served from the same domain is open in the browser). We should not encourage using it, it is there only to serve those cases where we cannot pass the information to the browser on where the favicon image resides (as in the PDF case).

Sort of like "last attempt".